File: Java/프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    int count = 0;
    public int solution(int[] numbers, int target) {
        int answer = 0;
        dfs(numbers, 0, target, 0);
        answer = count;
        return answer;
    }

    void dfs(int[] numbers, int depth, int target, int sum) {
        if (depth == numbers.length) {
            if (sum == target) {
                count++;
            }
            return;
        }
        dfs(numbers, depth + 1, target, sum + numbers[depth]);
        dfs(numbers, depth + 1, target, sum - numbers[depth]);
    }
}

File: Python/프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    answer = 0

    def dfs(depth, sum):
        nonlocal answer
        if depth == len(numbers):
            if sum == target:
                answer += 1
            return
        dfs(depth + 1, sum + numbers[depth])
        dfs(depth + 1, sum - numbers[depth])

    dfs(0, 0)

    return answer